### ISSUE TITLE
Unusable Errors/warnings and Conflict messages

### ISSUE THREAD

**Issue by romulus010689** [nextcloud/desktop#1195]
Error exists with Final Version 2.5.1. The Client shows a yellow waring message, "Conflicts exist, click here for details" or in German "Es existieren ungelöste Konflikte, für Details klicken".

However it is not possible to click somewhere on the yellow bar (at least nothing happens). The log is also not helpful to debug the messages.

I am using OSX Version 10.13.6.

Hope this could be fixed, since it is annoying to not find out which files are conflicted.

### Expected behaviour
The Button should show a list of files/paths on the machine which are conflicted, this way people can take a look which version they would keep.

### Actual behaviour
Nothing happens if the yellow bar is clicked.

### Steps to reproduce
see above.

### Client configuration
Client version: 2.5.1

Operating system: OSX High Sierra 10.13.6

OS language: German

Qt version used by client package (Linux only, see also Settings dialog): No Linux

Client package (From Nextcloud or distro) (Linux only): No Linux

Installation path of client:
No Idea, since I used the dmg or installer and OSX is not quiet transparent for me in this point, here I miss linux ;(

**Comment by vitaliis** [comment]
I was about to create new issue, but details are quite similar to what I observe.

For me the click on "There are unresolved conflicts. Click for details." works... but results in nonsense! :(
At best I would like to **see the relevant messages selected** in the "Activity"-view. This would greatly improve user experience and really help to resolve those conflicts. Instead of pointing to a pile of random messages, most of which in my case are either hidden or ignored.

**Comment by snowflakeas** [comment]
I'll add myself to this list - the current report of unresolved conflicts does not provide a good (any?) method of seeing which files are in conflict, or easily navigating to them.

**Comment by mlo-poc** [comment]
I want to add the Linux-Client (currently using 2.5.3) to this claim.

In general, the activity list is virtually unusable because it can't be filtered, or else configured in any way.

**Comment by zippytiff** [comment]
i was seeing this today, there almost needs to be an action for each conflict or for the whole list e.g

- client file wins
- add to exceptions list
- ignore file

HELP ! this has been such and issue with nextcloud for many years now, since the (from memory) 2.3 client

**Comment by romulus010689** [comment]
It still exists in 2.6.1 stable and meanwhile it seems to become a problem becaus sync is also disturbed somehow.. may there is any kinde of a workaround to at least resolve those conflicts? I am not able to search them manually since its a big amount of files.

**Comment by SpamReceiver** [comment]
I'd like to add the Windows client, too.
It shows the same unusable behaviour.

**Comment by veddox** [comment]
@er-vin I'm thinking this is an oversight from the last big redesign - IIRC, clicking the yellow/red notification bar used to open the activities tab in the settings window. Except, of course, that activities tab no longer exists.

Is there already a specific plan for what a new solution should look like?

(Also, on a side note, the activities pane itself could *really* do with a filter function - is anything in that direction being discussed somewhere?)

**Comment by er-vin** [comment]
@jancborchardt opinions welcome there

**Comment by vasyugan** [comment]
> @jancborchardt opinions welcome there

My opionion, or rather my observation is that 95% of my users simply ignore conflicts. Sure, the nextcloud icon has changed to yellow, but they don't know what this means and are simply overwhelmed with the task of figuring it out and solving the conflict. 

Therefore, the Nextcloud desktop client needs to offer some interactive dialogue that pops up when a conflict has been detected and offers some simple solutions, such compare the dates and sizes of the conflict file with the file that has been downloaded and ask the user to decide which one to keep and which one to delete. 

I guess anything beyond that such as offering to show and merge differences is not realistic for most file types and would be something that  could maybe be added later on for some file types were this makes sense. You definitely can't do that for multimedia files but also not for binary office documents, the latter accounting for most conflicts, in my observation.

Anyway, the goal should really be a workflow that prevents those conflicts to just go unnoticed/ignored for forever as they currently do. Again, I would suggest an alert which offers simple conflict resolution as described above (with the option "remind me later", for those who still don't dare to decide which file to keep and which to delete)

**Comment by HackintoshHD** [comment]
I would appreciate if fixing this design error could receive a higher priority on the desktop client's roadmap. It has become a real 'deal breaker' for the Desktop Client versions ≥ 2.5.0 insofar as everybody with the slightest conflict in his/her sync history will have to revert to version 2.3.3.84 to preserve a proper synchronization status.

**Comment by jjjpt** [comment]
I've had the same issue for quite a while (see also https://github.com/nextcloud/desktop/issues/1080)

As of 3.0.1 with the Windows client, clicking on the taskbar Nextcloud icon shows a list of recent activities + appears to flag pending conflicts at the top of the list.

**Comment by rasos** [comment]
The 3.0.1 client shows a list of last activities including conflicts, however with a longer file path it is not readable, which file is affected. And the yellow badge is not clickable. So users are left alone with some warnings and they can't track them down. This makes them feel that something is wrong and they don't know what to do. 

In general, the wording "conflicts" sounds very dangerous to end users. I would suggest to rephrase `conflicting files` to `branched version`, which is more clear what the issue is.

**Comment by jancborchardt** [comment]
> Therefore, the Nextcloud desktop client needs to offer some interactive dialogue that pops up when a conflict has been detected and offers some simple solutions, such compare the dates and sizes of the conflict file with the file that has been downloaded and ask the user to decide which one to keep and which one to delete.

@vasyugan completely agree there.

- In the case of conflicts, the icon needs to change of course, like now
- Conflicts should show on the very top of the activity list in the new main dialog / tray menu
- The entry should look like this:

> **Filename.pdf**
> Conflicts between local and server version

And multiple conflicts should be grouped:

> **Filename.pdf, image.png, and 4 more**
> Conflicts between local and server version

And to the right a button "Resolve conflict" / "Resolve conflicts" to make it obviously actionable. ("Resolve" only if we don’t have enough space which is probably the case.)

Clicking the entry or the button (both) should open a 2-step dialog like on Windows, see screenshot at https://answers.microsoft.com/en-us/windows/forum/all/do-this-for-all-conflicts-when-copying-moving/b125c81c-c0bd-4f3c-ad92-c62d8bf773a2

1. Simple dialog which offers the options to:
    - Use the local version
    - Use the server version
    - Compare details for the files (which goes to step 2)
2. Compare details for the files, which shows the dialog as in [Windows](https://answers.microsoft.com/en-us/windows/forum/all/do-this-for-all-conflicts-when-copying-moving/b125c81c-c0bd-4f3c-ad92-c62d8bf773a2) (we have it in our Android app as well).

**Comment by stevleibelt** [comment]
Hey, just got this issue today first time. 

![image](https://user-images.githubusercontent.com/2287220/94061081-2f3bae80-fde5-11ea-99d9-bbc63e6e0913.png)

I am running an Arch Linux with i3-wm as window manager. No taskbar or anything else, just usable screen place.
Is there any log file available I can tail or any information so I can solve the conflict on my own?
Or is there any generic walk around?

Plus and most important, any idea when this issue could be fixed?

Thanks in advance and especially for your work so far.

**Comment by cyclops1982** [comment]
Hi,

I have the same "issue". 

I honestly believe this is a user-experience issue. My path's are very deep and thus i just can't see where the files with conflicts are. If we simply allow us to click on the conflict item in the dialog that shows up when you press the icon and that click would take you to the folder location - that would already be a great step forward.

The  phone app has a nice dialog (bit buggy as well) around selecting which file you want. 
It would also be good to specify some standard/default file resolution option. For example, a file always has a "last modified date" and this it would be great to have an option where you can say "just always take the latest". The desktop client doesn't seem to have options for that.

**Comment by er-vin** [comment]
> I honestly believe this is a user-experience issue. My path's are very deep and thus i just can't see where the files with conflicts are. If we simply allow us to click on the conflict item in the dialog that shows up when you press the icon and that click would take you to the folder location - that would already be a great step forward.

That's 3.0.2 is supposed to do though. There was a bug in 3.0.1 and before which led to the wrong place but that should be gone now.

**Comment by Bockeman** [comment]
Nextcloud-Client 3.0.2 on Windows 10
"Click for details" does nothing.

**Comment by cyclops1982** [comment]
> 
> 
> > I honestly believe this is a user-experience issue. My path's are very deep and thus i just can't see where the files with conflicts are. If we simply allow us to click on the conflict item in the dialog that shows up when you press the icon and that click would take you to the folder location - that would already be a great step forward.
> 
> That's 3.0.2 is supposed to do though. There was a bug in 3.0.1 and before which led to the wrong place but that should be gone now.

Ah great. I didn't get a notification on 3.0.2 being available (even after 'check for update'). Anyway - i downloaded the installer from the site and when i click on the conflict i now go to the file location, which is great!

**Comment by cyclops1982** [comment]
> Nextcloud-Client 3.0.2 on Windows 10
> "Click for details" does nothing.

I have the same in 3.0.2. If you click once on the icon, then you get that "Main dialog" (right click on the icon -> Main Dialog" does the same). Then click on the item there and you will go to the folder in explorer and fix your conflict.

**Comment by Bockeman** [comment]
@cyclops1982 Thanks for commenting.
For clarity, my problem with 3.0.2 is that clicking on the yellow bar containing "Click for details" does nothing.  No dialog box appears.

Although there may be a "user experience issue" with what to do with the information were a dialog box to appear (which it does not), or how to find the conflicting files by other means, this has nothing to do with a clear instruction on the yellow bar which does nothing when you follow those instructions.  My user experience issue is simply: it does not do what it says on the tin.

**Comment by cyclops1982** [comment]
> 
> 
> @cyclops1982 Thanks for commenting.
> For clarity, my problem with 3.0.2 is that clicking on the yellow bar containing "Click for details" does nothing. No dialog box appears.
> 
> Although there may be a "user experience issue" with what to do with the information were a dialog box to appear (which it does not), or how to find the conflicting files by other means, this has nothing to do with a clear instruction on the yellow bar which does nothing when you follow those instructions. My user experience issue is simply: it does not do what it says on the tin.

I confirm that this issue still exist.

**Comment by shellstrom** [comment]
Just tried with Nextcloud 3.0.2 on Windows 10, and I can confirm this problem exists. Yellow exclamation mark and nothing happens when clicking "Click here for details."

**Comment by felixbinder** [comment]
On macOS with 3.0.2 clicking on "Click here for details" does nothing as well.

**Comment by vatolin** [comment]
Here the same at macOS 10.13.6.

Is there, as a workarround, anywhere a logfile or something else in order to learn about the conflict and to manually solve it? The yellow menuelet is annoying to me.

**Comment by rasos** [comment]
Even worse, the `Settings` view has no more activity tab per server. When I have seen some conflicts there is no chance to track them if you are syncing one or more nextclouds with heavy activity. The shrinked desktop widget log is not sufficient to analyse conflicts.

**Comment by eppfel** [comment]
The issue persists on my machines: 3.0.2 on MacOS 10.14.6 and MacOS 10.15.6

ATM there is no way to resolve sync conflicts. File-sync is the core functionality of nextcloud.  I hope this issue gets a higher priority.

**Comment by Alcatraz077** [comment]
I can confirm this issue still persists on Windows 10 with the 3.0.2 client.

**Comment by er-vin** [comment]
You will want to test again with 3.1.0 when it comes out. If you can't wait you will be able to test with one of the dailies after the PR referenced a few days ago got merged (which will close this issue ticket).

**Comment by iGadget** [comment]
3.0.3 on Ubuntu 20.10 - nothing is showing when clicking 'Click for details.'. Is there any scheduled / aimed release date for 3.1.0?

**Comment by er-vin** [comment]
We're aiming at having a first prerelease for 3.1 this week.

**Comment by holocronweaver** [comment]
Clicking the "Conflicts exist, click here for details" button still doesn't work in 3.4.1 on Ubuntu 20.04.

**Comment by Bockeman** [comment]
@holocronweaver, as fellow a user, I am curious.

I'm still on client 3.4.0.  _(Aside: The check for updates buttom is no longer present, so I just have to wait until my clients get updated through the new, more automatic update mechanism.)_

I have not seen any problems when clicking on "Conflicts exist, click here for details" button on many versions up to 3.4.0.  If you are seeing a specific problem with 3.4.1, I suggest you open a new issue because this isssue, as originally reported, has been fixed, the issue has been closed.  You may be seeing something new.

Of much more interest to me is why you even see a conflict.  Although I use the  "Conflicts exist, click here for details" button to get a quick overview, typically I run a script that contains:
```
find /mnt/<client_whatever> -name \*conflicted\*
```
to give me a complete picture of all conflicts.  (Hence you can still get the information even if the button is not working for you. And you can fix the conflict by either removing the conflicted file [after backing it up, just in case], or renaming it to replace the current file [after backing that up, just in case],)

If you have only one client that is changing files in any given folder tree, then you should never get conflicts.  There is an ongoing open issue and plenty of discussion on #2467.  If you are observing similar, please comment there.

**Comment by marbx** [comment]
Conflict notification: Clicking on "For details click here" does not do anything

Ubuntu 20.04.4 LTS

sudo apt install nextcloud-desktop
nextcloud-desktop is already the newest version (3.4.3-20220306.150807.99ab28eeb-1.0~focal1)

Workaround: `find . -iname *conflict* -type f`


Could you please reopen this issue, or create one specific to Linux?

**Comment by new-on-github** [comment]
> Conflict notification: Clicking on "For details click here" does not do anything
> [...]
> 
> Could you please reopen this issue, or create one specific to Linux?
Same here on
Linux Mint 20.3 Cinnamon (5.2.7)
Client-Version 3.5.0

Btw thanks for the work

**Comment by Xylon2** [comment]
yep would definitely be nice if it actually tells me what the conflicted files are

**Comment by iGadget** [comment]
Version 3.5.4 here on Pop!_OS 22.04 and it's *still* giving me this BS from time to time. Definitely not fixed.

**Comment by calidancer** [comment]
Today I have updated and restored from backup my AIO version of NextCloud 26.0.2 and after messing a while I noticed this issue:"Unresolved conflicts, click for details".
These are the steps that helped me to resolve them:

1. Stop your NextCloud client
2. Find the file **.nextcloudsync.log** in the NextCloud folder and move it "away" in a different folder which is not part of any synchronization (I have deleted it)
3. Relaunch your NextCloud client and let it do the scan before starting the synchronization
4. It should come back with the same error message but now if you click on the NextCloud logo and open it you should see the **LIST OF CONFLICTS** (apologize for my uppercase):
![image](https://github.com/nextcloud/desktop/assets/113770442/5e91ae94-c26a-4c4c-ad50-c6a47550f231)
5. Resolve your conflicts now and enjoy the green logo of your NextCloud instance

**Comment by locksec** [comment]
This is now happening on Windows 11, client 3.14.1. I can't click the yellow tray icon or the yellow bar, nothing happens. I have to manually find the .conflicted file and resolve manually.

**Comment by trya2l** [comment]
Although, I'm still facing this annoying issue on Fedora Linux (Nextcloud Desktop Client from Flathub v4.0.1), what worked for me to find the culprit file was

```sh
# in the nextcloud folder
find | grep conflicted
```

**Comment by tootbrute** [comment]
I have this error in the flatpak. My OS is Bluefin (Version: 43.20260505). I tried reporting in the flatpak repo, they sent me here.

Nextcloud Desktop Client Version 33.0.4daily (sic) (KDE running on Wayland) flatpak.

**Comment by marbx** [comment]
search for "conflicted" files.
